### PR TITLE
Fix end of game antag roles

### PR DESCRIPTION
**Pull request by flappybatpal**
[BUG]

## About the PR 
Fixes critter antags showing as blank.
Gives poltergeist special role and adds to agimmicks.
Rewrites bad_traitorify and use it to handle antag critters.
Stops some antag roles showing up twice on the scoreboard?
Removes objectives from hostile critters as a bunch of objectives make little sense for them/could be impossible.

## Why's this needed? 
Scoreboard should display roles.

### CODE REVIEW

**Comment by flappybatpal** [comment]
It's very likely some of this is bad as it took a lot of trial and error, please advise.

File: code/datums/abilities/wraith.dm [hunk 1]
```diff
@@ -728,6 +728,10 @@
 		lucky_dude.special_role = "poltergeist"
 		lucky_dude.dnr = 1
 		lucky_dude.transfer_to(P)
+		if (!lucky_dude.former_antagonist_roles.Find(lucky_dude.special_role))
```
Comment (ZeWaka) [inline code/datums/abilities/wraith.dm:731]: use `in`, you don't need Find's idx

File: code/modules/antagonists/changeling/abilities/hivemind.dm [hunk 2]
```diff
@@ -90,6 +90,8 @@
 				spider.mind.special_role = "handspider"
 			if (!(spider.mind in ticker.mode.Agimmicks))
 				ticker.mode.Agimmicks += spider.mind
+			if (!spider.mind.former_antagonist_roles.Find(spider.mind.special_role))
```
Comment (ZeWaka) [inline code/modules/antagonists/changeling/abilities/hivemind.dm:93]: in

File: code/modules/antagonists/changeling/abilities/hivemind.dm [hunk 3]
```diff
@@ -182,6 +184,8 @@
 				spider.mind.special_role = "eyespider"
 			if (!(spider.mind in ticker.mode.Agimmicks))
 				ticker.mode.Agimmicks += spider.mind
+			if (!spider.mind.former_antagonist_roles.Find(spider.mind.special_role))
```
Comment (ZeWaka) [inline code/modules/antagonists/changeling/abilities/hivemind.dm:187]: in

File: code/modules/antagonists/changeling/abilities/hivemind.dm [hunk 4]
```diff
@@ -275,6 +279,8 @@
 				spider.mind.special_role = "legworm"
 			if (!(spider.mind in ticker.mode.Agimmicks))
 				ticker.mode.Agimmicks += spider.mind
+			if (!spider.mind.former_antagonist_roles.Find(spider.mind.special_role))
```
Comment (ZeWaka) [inline code/modules/antagonists/changeling/abilities/hivemind.dm:282]: in

File: code/modules/antagonists/changeling/abilities/hivemind.dm [hunk 5]
```diff
@@ -359,6 +365,8 @@
 				crab.mind.special_role = "buttcrab"
 			if (!(crab.mind in ticker.mode.Agimmicks))
 				ticker.mode.Agimmicks += crab.mind
+			if (!crab.mind.former_antagonist_roles.Find(crab.mind.special_role))
```
Comment (ZeWaka) [inline code/modules/antagonists/changeling/abilities/hivemind.dm:368]: in

**Review by ZeWaka — CHANGES_REQUESTED** [review]

**Comment by flappybatpal** [comment]
Pruned the former_antagonist_roles additions. From looking at it, this did not seem suitable for changeling limbs and would be a new addition.
Changed the poltergeist to be added to agimmicks, looks like they weren't tracked as being an antag at all.

Changed the tips popup for generic antag.

File: code/procs/jobprocs.dm [hunk 6]
```diff
@@ -730,22 +730,33 @@
...
+			var/select_objective = pick(eligible_objectives)
+			new_objective = new select_objective
+			new_objective.owner = H.mind
+			new_objective.set_up()
+			H.mind.objectives += new_objective
+			H << browse(grabResource("html/traitorTips/traitorhardTips.html"),"window=antagTips;titlebar=1;size=600x400;can_minimize=0;can_resize=0")
+		if(!(H.mind in ticker.mode.traitors))
+			ticker.mode.traitors += H.mind
+	else
+		if (!(H.mind in ticker.mode.Agimmicks))
```
Comment (ZeWaka) [inline code/procs/jobprocs.dm:753]: |=

File: code/procs/jobprocs.dm [hunk 7]
```diff
@@ -730,22 +730,33 @@
...
+		var/num_objectives = rand(1,3)
+		var/datum/objective/new_objective = null
+		for(var/i = 0, i < num_objectives, i++)
+			var/select_objective = pick(eligible_objectives)
+			new_objective = new select_objective
+			new_objective.owner = H.mind
+			new_objective.set_up()
+			H.mind.objectives += new_objective
+			H << browse(grabResource("html/traitorTips/traitorhardTips.html"),"window=antagTips;titlebar=1;size=600x400;can_minimize=0;can_resize=0")
+		if(!(H.mind in ticker.mode.traitors))
```
Comment (ZeWaka) [inline code/procs/jobprocs.dm:750]: |=

File: code/datums/abilities/wraith.dm [hunk 8]
```diff
@@ -728,6 +728,8 @@
 		lucky_dude.special_role = "poltergeist"
 		lucky_dude.dnr = 1
 		lucky_dude.transfer_to(P)
+		if (!(lucky_dude in ticker.mode.Agimmicks))
```
Comment (ZeWaka) [inline code/datums/abilities/wraith.dm:731]: |=

**Pull request merged**